android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/authorization/AuthorizationsDao.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2022 Solana Mobile Inc.
+ */
+
+package com.solana.mobilewalletadapter.walletlib.authorization;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteStatement;
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*package*/ class AuthorizationsDao extends DbContentProvider<AuthRecord>
+        implements AuthorizationsSchema, PublicKeysSchema,
+        WalletUriBaseSchema, AuthorizationsDaoInterface {
+
+    AuthorizationsDao(SQLiteDatabase db) {
+        super(db);
+    }
+
+    @Override
+    protected AuthRecord cursorToEntity(Cursor cursor) {
+        return null;
+    }
+
+    private AuthRecord cursorToEntity(Cursor cursor, @NonNull IdentityRecord identityRecord, long authorizationValidityMs) {
+        final int id = cursor.getInt(0);
+        final long issued = cursor.getLong(1);
+        final int publicKeyId = cursor.getInt(2);
+        final int walletUriBaseId = cursor.getInt(3);
+        final byte[] scope = cursor.getBlob(4);
+        final String cluster = cursor.getString(5);
+        final byte[] publicKey = cursor.getBlob(6);
+        final String accountLabel = cursor.isNull(7) ? null : cursor.getString(7);
+        final Uri walletUriBase = cursor.isNull(8) ? null : Uri.parse(cursor.getString(8));
+        return new AuthRecord(id, identityRecord, publicKey,
+                accountLabel, cluster, scope, walletUriBase, publicKeyId, walletUriBaseId,
+                issued, issued + authorizationValidityMs);
+    }
+
+    @Override
+    public long insert(int id, long timeStamp, int publicKeyId, String cluster, int walletUriBaseId, byte[] scope) {
+        final ContentValues reissueContentValues = new ContentValues(6);
```

Comment:
NIT: the name `resissueContentValues` is no longer relevant without the surrounding context. I'd rename it to just `contentValues`